modules/monixBio/test/src/weaver/monixbiocompat/IOSuiteTest.scala

```diff
@@ -0,0 +1,58 @@
+package weaver.monixbiocompat
+
+import weaver.framework.DogFood
+
+import monix.bio.Task
+import sbt.testing.Status
+
+object IOSuiteTest extends SimpleIOSuite {
+  List(
+    TestWithExceptionInTest,
+    TestWithExceptionInExpectation,
+    TestWithExceptionInInitialisation
+  ).foreach { testSuite =>
+    test(s"fail properly in ${testSuite.getClass.getSimpleName}") {
+      for {
+        (_, events) <- DogFood.runSuite(testSuite).to[Task]
+      } yield {
+        val event = events.headOption.get
+        expect(event.status() == Status.Error) and
+          expect(event.throwable().get().getMessage == "oh no")
+      }
+    }
+  }
+
+  test("fail properly on failed expectations") { _ =>
+    for {
+      (_, events) <- DogFood.runSuite(TestWithFailedExpectation).to[Task]
+    } yield expect(events.headOption.get.status() == Status.Failure)
+  }
+
+  object TestWithExceptionInTest extends SimpleIOSuite {
+    test("example test") {
+      Task.raiseError(new RuntimeException("oh no"))
+    }
+  }
+
+  object TestWithExceptionInExpectation extends SimpleIOSuite {
+    test("example test") {
+      for {
+        _ <- Task.unit
+      } yield throw new RuntimeException("oh no")
```

Comment:
@avasil, 👋 , pinging your for advice. We're adding support for monix-bio, and this throw seems to make it hang forever. Is it to be expected, or is it a bug ? 

Now I'm aware this is very much the fault of the user, but being a maintainer of a test framework, and knowing how liberal people are with tests in general, I know many will likely mess up and come to us for explanation 😅 

Is there anything on our side we could do to protect against it ?

Reply:
Apologies, wrong analysis, the problem seem to be lying elsewhere

Reply:
@avasil, actually I think my first analysis was correct : monix-bio seems to get stuck when throws are performed in a `map` call, even when the error channel expect throwable. 

Any guidance on how to recover would be appreciated

Reply:
Yes, if there is an error in `map`, it is treated as an unexpected error (in a different channel). 
It can catch off guard when using `IO[Throwable, A]`

I assume that it hangs because the error is not handled properly - you can use `redeemCause` to get access to both types of errors. If you are able to override `test` method, you could do `redeemCauseWith(c => IO.raiseError(c.toThrowable), x => IO.now(x))`

I think I should add an operator that changes all terminal errors to typed errors if the type is `Throwable`, like ZIO's absorb.

Reply:
@Baccata I provided an implementation of `Test.apply` in the Monix BIO compat module with the additional error checks @Avasil mentioned and this seemed to resolve the issue. Commit incoming. 

It required minimal changes in core code, but please tell me if there is a better way to go about it.